stage0/src/zero_page.rs

```diff
@@ -248,7 +250,14 @@ impl ZeroPage {
     }
 
     /// Updates the pointer to the command line parameter string in the zero page.
-    pub fn set_cmdline(&mut self, cmdline: &'static CStr) {
+    pub fn set_cmdline<T: AsRef<str>>(&mut self, cmdline: T) {
+        let source = cmdline.as_ref().as_bytes();
+        // Create enough space to include a terminating \0;
+        let mut buf = Vec::with_capacity_in(source.len() + 1, &BOOT_ALLOC);
+        buf.resize(buf.capacity(), 0u8);
+        buf[..source.len()].copy_from_slice(source);
```

Comment:
Use `CString` here instead of dealing with the null termination manually: https://doc.rust-lang.org/stable/std/ffi/struct.CString.html#method.new

Reply:
Good idea. Done.